microsoft/testsuites/core/hv_module.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from assertpy import assert_that
+
+from lisa import Logger, RemoteNode, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import CentOs, Redhat
+from lisa.tools import Dmesg, Lsmod, Modinfo, Modprobe, Rpm, Uname
+from lisa.util import SkippedException
+
+
+@TestSuiteMetadata(
+    area="core",
+    category="functional",
+    description="""
+    This test suite covers test cases previously handled by LISAv2:
+    LIS-MODULES-CHECK, VERIFY-LIS-MODULES-VERSION,
+    INITRD-MODULES-CHECK, RELOAD-MODULES-SMP
+    """,
+)
+class HvModule(TestSuite):
+    def get_modules(self, node: RemoteNode) -> list[str]:
+        """
+        Returns the hv_modules that are not directly loaded into the kernel.
+        """
+        hv_modules = [
+            "hv_storvsc",
+            "hv_netvsc",
+            "hv_vmbus",
+            "hv_utils",
+            "hid_hyperv",
+            "hv_balloon",
+            "hyperv_keyboard",
+        ]
+        uname = node.tools[Uname]
+        kernel_version = uname.get_linux_information().kernel_version_raw
+        config_path = f"/boot/config-{kernel_version}"
+
+        if node.execute(f"grep CONFIG_HYPERV_STORAGE=y {config_path}").exit_code == 0:
+            hv_modules.remove("hv_storvsc")
```

Comment:
1. A loop looks better.
2. The logic can be more straightforward, if it add an item in if it exists, instead of deleting.